src/pages/data.js

```diff
@@ -1,113 +1,107 @@
-    }
-    /* incoming response { pagination: {}, data: [] } */
-    let { data: { data }, error } = useSWR(url, swrOptions);
-    if (error) return <div>failed to load</div>;
-    return(
-        <>
-            <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', marginTop: 20 }}>
-                <div style={{ width: '95%', maxWidth: 1000}}>
-                    <h2>Explore the data on your own</h2>
-                    <p>Change the units using the preference modal from the navbar. </p>
-                </div>
-                <CustomDataGrid data={data} preferences={preferences}/>
-            </div>
-
-            <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', marginTop: 20 }}>
-
-                <div style={{ width: '95%', maxWidth: 1000}}>
-                    <h2>See how the data has changed over time</h2>
-                    <h3 style={{marginTop: 3, marginBottom: 3}}>Show data for:</h3>
-
-                    <div style={{display: 'flex'}}>
-                        {dateRanges.map((r, idx) => (
-                            <div key={idx}
-                                style={{padding: '5px 0', marginRight: 15, borderBottom: r.active ? '3px solid #1565c0' : ''}}
-                                onClick={() => {
-                                    setDateRange(prevRange => {
-                                        prevRange.active = false;
-                                        r.active = true;
-                                        return r;
-                                    });
-                                }}
-                            >
-                                <p style={{color: '#1565c0', padding: 0, margin: 0, cursor: 'pointer'}}>{r.label}</p>
-                            </div>
-                        ))}
-                    </div>
-
-                    <h3 style={{marginBottom: 0}}>Select data to show:</h3>
-                </div>
-
-                <CustomAreaChart data={data}/>
-            </div>
-        </>
-    );
+export default function App({ initialData }) {
+  const { preferences } = useContext(PreferenceContext);
+
+  /* define how long period should show */
+  const [dateRange, setDateRange] = useState(() => {
+    const range = dateRanges[7]; // max
+    range.active = true;
+    return range;
+  });
+
+  let url = urlWithParams(endpointUrl, {
+    temperature_unit: preferences.temperature_unit.symbol,
+    conductivity_unit: preferences.conductivity_unit.symbol,
+    location_name: preferences.location.symbol,
+    start_date: dateRange.startDate.toISOString(),
+    page_size: 2000,
+  });
+
+  const swrOptions = {
+    fetcher: () => fetcher(url),
+    fallbackData: { pagination: {}, data: [] },
+    refreshInterval: 1000 * 60,
+  };
+  /* incoming response { pagination: {}, data: [] } */
+  let { data: { data }, error } = useSWR(url, swrOptions);//data for CustomAreaChart, but maybe let comp get data?

```

Comment:
will be fixed in branch [ws60](https://github.com/sensor-network/open-data-portal/tree/ws60)